apps/emqx_rule_engine/src/emqx_rule_engine.erl

```diff
@@ -216,8 +216,14 @@ get_rules_with_same_event(Topic) ->
 get_rule_ids_by_action(ActionName) when is_binary(ActionName) ->
     [
         Id
-     || #{actions := Acts, id := Id} <- get_rules(),
-        lists:any(fun(A) -> A =:= ActionName end, Acts)
+     || #{actions := Acts, id := Id, from := From} <- get_rules(),
+        lists:any(fun(A) -> A =:= ActionName end, Acts) orelse
+            lists:any(
```

Comment:
Think it would help to wrap this in a function with a name that explains what it does

Reply:
:heavy_check_mark: